src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -147,6 +148,31 @@ void remove_redundant_reorders::run(program& p) {
         }
     }
 
+    // Remove redundant reorder in dyanmic shape for an user node which perfers same format
+    itr = p.get_processing_order().begin();
+    while (itr != p.get_processing_order().end()) {
+        auto& node_ptr = *itr++;
+        if (!node_ptr->is_type<reorder>() || !node_ptr->is_in_data_flow() || node_ptr->get_users().size() != 1 ||
+            node_ptr->get_dependencies().size() != 1)
+            continue;
+
+        auto& node = node_ptr->as<reorder>();
+        auto& usr = node_ptr->get_users().front();
+        auto& dep = node_ptr->get_dependency(0);
+
+        auto redundant_format = usr->is_type<resample>() &&
```

Comment:
As we've already discussed in Team Chat, we need to find the root cause of why resample was unnecessarily set to planar in the first place. This is because unnecessary reordering was added, and logic to remove it was added here.

And if reordering is unavoidable, wouldn't there be a related part in the existing remove_redundant_reorders? Because removing reordering when dep and user have the same format seems to be a common pattern NOT specific resample.